scripts/collect_ad_boilerplate.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import re
+
+import logging
+
+
+def write_detector_class(boilerplate_file, dev_name, det_name, cam_name):
+    """
+    Writes boilerplater 'Detector' class for ophyd/areadetector/detectors
+    """
+
+    boilerplate_file.write(
+        f'''
+class {det_name}(DetectorBase):
+    _html_docs = ['{dev_name}Doc.html']
+    cam = C(cam.{cam_name}, 'cam1:')
+''')
+
+
+def parse_pv_structure(driver_dir):
+    """
+    Reads all .template files in the specified driver directory and maps them
+    to the appropriate EPICS signal class in ophyd
+
+    Also determines if the Cam class should extend the FileBase class as well.
+    """
+
+    # Find the template directory following the standard areaDetector project format
+    template_dir = driver_dir
+    for dir in os.listdir(driver_dir):
+        if dir.endswith('App'):
+            template_dir = os.path.join(template_dir, dir, 'Db')
+            break
+    logging.debug(f'Found template dir: {template_dir}')
+
+    # Create a list of file paths to all template files.
+    template_files = []
+    for file in os.listdir(template_dir):
+        if file.endswith('.template'):
+            template_files.append(os.path.join(template_dir, file))
+            logging.debug(f'Found template file {file}')
+
+    # Dict mapping pv name to appropriate EPICS signal, based on typical
+    # PV and PV_RBV format for areaDetector
+    pv_to_signal_mapping = {}
+    include_file_base = False
+
+    for file in template_files:
+        logging.debug(f'Collecting pv info from {os.path.basename(file)}')
+        fp = open(file, 'r')
+
+        lines = fp.readlines()
+        for line in lines:
+
+            # If NDFile.template is included, we need to extend FileBase as well.
+            if line.startswith('include "NDFile.template"'):
+                logging.debug(f'Driver AD{dev_name} uses the NDFile.template file.')
+                include_file_base = True
+
+            # identify any lines that start with 'record
+            if line.startswith('record'):
+                # Get the name of the PV.
+                pv_name = line.split(')')[2][:-1]
+
+                # Check if it is a readback PV
+                if pv_name.endswith('_RBV'):
+                    # If it has a partner PV, switch the signal to SignalWithRBV
+                    if pv_name[:-4] in pv_to_signal_mapping.keys():
+                        logging.debug(f'Identified {pv_name[:-4]} as a record w/ RBV')
+                        pv_to_signal_mapping[pv_name[:-4]] = 'SignalWithRBV'
```

Comment:
`pv_name[:-4]` was used 3 times here. I think it is worth defining it once and reuse.

If you want to go fancy and be more general, you can get the length of `_RBV`, and use that, such as:
```py
rbv_suffix = '_RBV'
pv_name_wo_rbv = pv_name[:-len(rbv_suffix)]
...
```